src/app/submission/sections/form/section-form.component.ts

```diff
@@ -228,8 +229,8 @@ export class SubmissionSectionFormComponent extends SectionModelComponent {
 
     const sectionDataToCheck = {};
     Object.keys(sectionData).forEach((key) => {
-      if (this.sectionMetadata && this.sectionMetadata.includes(key) && this.inCurrentSubmissionScope(key)) {
```

Comment:
@alexandrevryghem 

I think that instead to turning to use the section.data object it would be better to include into the sectionMetadata object the metadata needed for the relationchip. You can do it by changing the implementation of the [computeSectionConfiguredMetadata](https://github.com/DSpace/dspace-angular/blob/main/src/app/submission/sections/sections.service.ts#L498) in order to include not only the selectableMetadata but also the selectableRelationship

This is because the section.data object could contain also metadata that are related to a different section form of the current submission which should not be considered by this method

Reply:
@alexandrevryghem 

could you commit the changes you mentioned [here](https://github.com/DSpace/dspace-angular/pull/2252#issuecomment-1581096673) to a different branch? so i can take a look.
The point is that with this change now we are changing the current behaviour because, as already said, this method should consider only metatada that belong only to the current submission section

Reply:
@atarix83 we've updated the "metadata to be checked" to use `sectionData` instead of `this.sectionData.data` like before, but we can't really reinstate the `this.sectionMetadata` check
* if we try to add `relation.*` MDFs like you suggested, Relationships get duplicated in the form -- haven't been able to track down what causes this
* even if we fix this, the `this.sectionMetadata` check prevents _removed_ Relationships from triggering a form update
  * looks like fields _removed_ from `sectionData` aren't flagged as differences
  * without the check, the form update is triggered by an unexpected `dspace.entity.type` change in _another section of the form_. If we reinstate the check, removing Relationships stops working

So all in all: we've attempted to somewhat alleviate your concerns, but likely won't be able to fully solve all of the above bugs in time for 7.6